Update Makefile to give make run options to supply a MODE=local option

snehagunta opened the pull request:
### PR Template:

## Describe your changes

- We recently introduced a local-build option to run the application locally without doing a FIPS compliance check. 
- Updated the Makefile targets, run and run-help to allow specification of a MODE=local option so we can run the application using a non FIPS compliant build.

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

tonytheleg commented:
Thanks @snehagunta i meant to make updates to this after Jonathan pointed out similar issues in relations-api (https://github.com/project-kessel/relations-api/pull/334#discussion_r1920649740)

For relations i updated a few of the make targets to call `local-build` instead, im fine with either approach but we'll also want to update the `pr-check` target too for the same reason

snehagunta commented:
Updated the implementation to directly use `local-build` in the run, run-help and pr-check targets. Removed the conditional, as these commands are only used locally, hence local-build should be the only mode to be used.

tonytheleg reviewed (approved):
Thank you!

LGTM

tylercreller commented:
LGTM
